### PR TITLE
Refs [TASK] [36166] Handle search hero

### PR DESCRIPTION
**Pull request by thanhvinh2k**
## Related Tickets
- [#Task 36166: Handle search hero by name](https://edu-redmine.sun-asterisk.vn/issues/36166)
## WHAT
- Handle search hero by name
## Evidence (Screenshot or Video)
handleSearch
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/hero_01/ui/champion/ChampionAdapter.kt [hunk 1]
```diff
@@ -47,6 +47,16 @@ class ChampionAdapter(private val onItemClickListener: OnItemRecyclerViewListene
         }
     }
 
+    fun getCircleDifficulty(level: Int?): Int {
+        return when (level) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/ui/champion/ChampionAdapter.kt]:
```suggestion
    fun getCircleDifficulty(level: Int?) = return when (level) {
```

File: app/src/main/java/com/sun/hero_01/ui/search/SearchAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    fun updateData(heroes: MutableList<Hero>?) {
+        heroes?.let {
+            this.searchHeroes.clear()
+            this.searchHeroes.addAll(it)
+            notifyDataSetChanged()
+        }
+    }
+
+    fun getDifficulty(level: Int?): String {
+        return when (level) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/ui/search/SearchAdapter.kt]:
```suggestion
    fun getDifficulty(level: Int?) = return when (level) {
```

File: app/src/main/java/com/sun/hero_01/ui/search/SearchAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            in HeroDifficulty.EASY.diff -> HeroDifficulty.EASY.nameDiff
+            in HeroDifficulty.AVERAGE.diff -> HeroDifficulty.AVERAGE.nameDiff
+            in HeroDifficulty.HARD.diff -> HeroDifficulty.HARD.nameDiff
+            in HeroDifficulty.SEVERE.diff -> HeroDifficulty.SEVERE.nameDiff
+            else -> ""
+        }
+    }
+
+    fun getCircleDifficulty(level: Int?): Int {
+        return when (level) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/ui/search/SearchAdapter.kt]:
```suggestion
    fun getCircleDifficulty(level: Int?) = return when (level) {
```

File: app/src/main/java/com/sun/hero_01/ui/search/SearchAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        in HeroDifficulty.SEVERE.diff -> HeroDifficulty.SEVERE.nameDiff
+        else -> ""
+    }
+
+    fun getCircleDifficulty(level: Int?) = when (level) {
+        in HeroDifficulty.EASY.diff -> R.drawable.ic_easy
+        in HeroDifficulty.AVERAGE.diff -> R.drawable.ic_average
+        in HeroDifficulty.HARD.diff -> R.drawable.ic_hard
+        in HeroDifficulty.SEVERE.diff -> R.drawable.ic_severe
+        else -> 0
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/ui/search/SearchAdapter.kt]: e tham khảo về first Index và tạo 1 func để gọi ra. Hiện tại a đang thấy lặp code 2 lần

Comment (thanhvinh2k) [reply app/src/main/java/com/sun/hero_01/ui/search/SearchAdapter.kt]: vâng a

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**